### PR TITLE
Catalog of my things

### PR DESCRIPTION
**Pull request by PolinaStamenova**
### Catalog of my things

**Database:** 
- Create a schema.sql file with tables that is analogical to the structure of the classes in our app. 
- Add all properties and associations from the parent Item class as table columns to all tables based on the child classes.

**Logic:** 
- Add **_Item_** **class** used as a constructor for the classes:
  - _Book._
  - _Movie._
  - _Game._
  - MusicAlbum.
- **Methods** in class **_Item_**:
   - Add  _**can_be_archaived?**_   method to the _Item_ class that return true if  a current conditions is met or false if not.
   - Based on its return value the method _**move_to_archived**_ is updating the instance variable _@archived (boolean)_. 
- Add  _**can_be_archaived?**_  to the following classes: **_Book, Movie, Game, MusicALbum_** that overrides the method 
  in the parent class **_Item_**, based on different conditions.

- Add classes:
   - _Genre._
   - _Author._
   - _Source._
   - _Label._
- Add _**add_item**_ **method** to all the classes mentioned above, that shows the relationship between them and Item 
    class. Every created instance is added into the instance variable **_@items_** (_array_) by calling the method. 

**Tests:**
   - Create test for the following classes:
     - _Book._
     - _Movie._
     - _Game._
     - MusicAlbum.
     - _Genre._
     - _Author._
     - _Source._
     - _Label._

**UI:**
 - Create a **_main.rb_** file that will serve as our console app entry-point.
 - By runung the **_main.rb_** file:
    - Presenting the user a list of options to perform.
    - The users can choose an option.   
    - Have a way to quit the app.
 - The following options are available:
    - List all books.
    - List all music albums.
    - List all movies.
    - List of games.
    - List all genres.
    - List all labels.
    - List all authors.
    - List all sources.
    - Add a book.
    - Add a music album.
    - Add a movie.
    - Add a game.
    
**JSON**
 - The data is preserved by saving collections in .json files.

🤝 **_Contribution_** _with:_ 
  - @PolinaStamenova 
  - @hiromataba 

_Thank you for taking time to review our PR!_ 😃 🌵

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,6 +104,19 @@ If you are non a technical person you can download the zip file.
...
+
+👤 **Polina Stamenova**
+
+- GitHub: [@githubhandle](https://github.com/PolinaStamenova)
+- LinkedIn: [LinkedIn](https://www.linkedin.com/in/polina-stamenova-a60766112/)
+
+
 =======
 
 ## 🤝 Contributing
```
Comment (Emmyn5600) [inline README.md:117]: - [ ] In the readme file, you can remove the Live link because it directs to the wrong link

File: RSpec/spec/game_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    it 'returns #multiplayer equals "true"' do
+      expect(@game.multiplayer).to(eq(true))
+    end
+    it 'returns #last_played_at" matches "2000-10-10' do
+      expect(@game.last_played_at).to(eq('2000-10-10'))
+    end
+    it 'returns true if publish_date is more than 10 years' do
+      expect(@game.can_be_archived).to(eq(true))
+    end
+  end
```
Comment (Emmyn5600) [inline RSpec/spec/game_spec.rb]: - [ ] let us add the following test:
- when published over 10 years ago and last played over 2 years
- when published less than 10 years ago and last played over 2 years

File: RSpec/spec/genre_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    @item.genre = @genre
+  end
+
+  describe '#test methods of class genre' do
+    it 'returns #name matches "Terror"' do
+      expect(@item.genre.name).to(eq('Terror'))
+    end
+    it 'returns #@items[0] matches "@items"' do
+      expect(@genre.items[0]).to(eq(@item))
+    end
```
Comment (Emmyn5600) [inline RSpec/spec/genre_spec.rb:17]: - [ ] Let us add this as well ` it should add an item`

File: RSpec/spec/music_album_spec.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  describe '#test methods of class music_album' do
+    it 'returns #on_spotify equals "true"' do
+      expect(@music_album.on_spotify).to(eq(true))
+    end
+    it 'returns true if publish_date is more than 10 years' do
+      expect(@music_album.can_be_archived).to(eq(true))
+    end
+  end
```
Comment (Emmyn5600) [inline RSpec/spec/music_album_spec.rb]: - [ ] Let us test this  `when published less than 10 years ago and on Spotify`

File: RSpec/spec/source_spec.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+  describe '#test methods of class source' do
+    it 'returns #source name matches "Shop"' do
+      expect(@item.source.name).to(eq('Shop'))
+    end
+    it 'returns #@items[0] matches "@items"' do
+      expect(@source.items[0]).to(eq(@item))
+    end
+  end
+end
```
Comment (Emmyn5600) [inline RSpec/spec/source_spec.rb:19]: - [ ] All in all, we are missing the tests for an `item_spec`

- move_to_archive
- add_genre
- add_author
- add_source
- add_label

File: store/store.rb [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    movie_json = items_2_json(@movies)
+    set_path('./store/movies.json', movie_json) if movie_json.size.positive?
+  end
+
+  if @music_albums.size.positive?
+    music_album_json = items_2_json(@music_albums)
+    set_path('./store/music_albums.json', music_album_json) if music_album_json.size.positive?
+  end
+end
+
```
Comment (Emmyn5600) [inline store/store.rb]: - [ ] Optional
If any of your methods are longer than ~5 lines, the code could likely be cleaner. You'll want to extract logic into a smaller helper method. This is not a general rule but is applicable to a very logical task like this.

File: classes/item.rb [hunk 7]
```diff
@@ -0,0 +1,42 @@
+require 'date'
+
+class Item
+  attr_reader :genre, :author, :source, :label
+
+  # rubocop: disable all
+  def initialize(id, publish_date, archived = false)
```
Comment (Emmyn5600) [inline classes/item.rb]: - [ ] Why are you disabling the rubocop errors, you included the rubocop.yml file to help you to deal with and correct the rubocop errors however you are disabling all errors. Try to correct some errors if it doesn't fix all you can mention it.

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi,

Good job so far! 👏🏼 @hiromataba  , @Metsanpeitto   and @PolinaStamenova 

May I highlight the following:
- The video presentation is showing almost all the features of the project:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- The git-flow has been followed :heavy_check_mark:
- The repo looks professional:heavy_check_mark:
- Your project looks awesome:heavy_check_mark: 
- The database is well structured :heavy_check_mark: 
- The data are being stored in JSON files :heavy_check_mark: 
- The project is running successfully:heavy_check_mark: 
- Tests are passing successfully :heavy_check_mark: 

## STATUS: CHANGES REQUESTED :recycle:


### Changes requested.

- _Check the comment under the review_

 #### Optional suggestions
 
 - N/A
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: README.md (hunk 1 above)
Comment (Metsanpeitto) [reply README.md:117]: :heavy_check_mark: Done!

File: RSpec/spec/game_spec.rb (hunk 2 above)
Comment (Metsanpeitto) [reply RSpec/spec/game_spec.rb]: :heavy_check_mark: Done!

File: RSpec/spec/genre_spec.rb (hunk 3 above)
Comment (Metsanpeitto) [reply RSpec/spec/genre_spec.rb:17]: :heavy_check_mark: Done!

File: RSpec/spec/music_album_spec.rb (hunk 4 above)
Comment (Metsanpeitto) [reply RSpec/spec/music_album_spec.rb]: :heavy_check_mark: Done!

File: RSpec/spec/source_spec.rb (hunk 5 above)
Comment (Metsanpeitto) [reply RSpec/spec/source_spec.rb:19]: :heavy_check_mark: Done!

File: store/store.rb (hunk 6 above)
Comment (Metsanpeitto) [reply store/store.rb]: :heavy_check_mark: Done!

File: classes/item.rb (hunk 7 above)
Comment (Metsanpeitto) [reply classes/item.rb]: Sorry Sir, There is a rubocop conflict when using that syntax to parse the arguments. 

If I use this syntax 

```cmd
 archived = false 
```

The code works but I got a rubocop error and it suggests I use the following way :

```cmd
archived: false
``` 
Then I don't get rubocop error but the code doesn't work and complains about the number of arguments being not 3.

Comment (Metsanpeitto) [reply classes/item.rb]: :heavy_check_mark: Fixed.
I disabled it only for that concrete error.
```cmd

  # rubocop: disable Style/OptionalBooleanParameter
  def initialize(id, publish_date, archived = false)
```

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED ✔🎉**
Hello @PolinaStamenova 👋🏽
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**